source/manuals/programming-languages/python/python.html.md.erb

```diff
@@ -171,6 +171,15 @@ Note: you can also ignore rules on particular lines of code or files by adding a
 * [Flake8 plugins list][flake8-plugins]
 
 
+## Environments
+
+Use distinct Python environments for each project you work on.
+We recommend [Pyenv](https://github.com/pyenv/pyenv) to manage Python environments.
+
+We recommend using [direnv](https://direnv.net/) to manage environment variables. 
+This ensures project specific variables do not clutter your main environment or the environment of other projects.
+direnv uses `.envrc` for general project specific variables and you can use a non version controlled `.secrets` to store sensitive information.
```

Comment:
It's not clear to me why this is a Python specific recommendation... does direnv hook into pyenv in some way?

Reply:
I've never used it 😅

Reply:
just to clarify in comments as i'm also doing in the doc;

pyenv to manage the python version
pvenv-virtualenv to manage your virtual envs
direnv to manage environment variables although not specific to Python, included because it is still part of env set up